### DISCUSSION TITLE
general relationship of XML and source code relate to XML properties

### DISCUSSION THREAD

**Discussion by DKAMX** [FakeFishGames/Barotrauma#6696] [Code questions and modding support]
Hi, developers.
I am studying your source code for a period of time.
I was interested in those items defined in XML and what kind of elements and their property they have.
I look up the source code and have some progress, however, I do not completely understand the relation between the source code and XML.
For example, the XML `<RequiredItem />` it has some property such as `identifier` and `amount`, I found those properties also defined in source code. While the rest of the property such as `type` and `items` are not been found.
Could you explain some about this? Where did those properties being defined?
I am starting to make a sort of Documentation that records all kinds of XML elements and explains them, so players can make the mod easier.
Many thanks.

**Comment by 3e849f2e5c** [comment]
The source code is a bit messy in this regard, `<RequiredItem />` XML element is most likely `RelatedItem` class found in  `Items/RelatedItem.cs` while the `RequiredItem` class found in `Items/ItemPrefab.cs` is used for crafting recipes. I remember this caught me couple times too in the past.
